src/formats/yarle/utils/folder-utils.ts

```diff
@@ -119,7 +120,20 @@ export const clearResourceDir = (note: any): void => {
 	resourceDirClears.set(resPath, clears + 1);
 };
 
-export const setPaths = (enexSource: PickedFile): void => {
+export const handleNotebookstacks = (enex: PickedFile, options: YarleOptions): void => {
+	const notebookFolderNames = enex.basename.split('@@@');
+	let notebookName = notebookFolderNames.pop();
+	if (!notebookName){
+		notebookName = enex.basename;
+	}
+	fs.mkdirSync(path.join(options.outputDir, ...notebookFolderNames), { recursive: true });
+	enex.fullpath = enex.fullpath.replace(enex.basename, notebookName || enex.basename);
```

Comment:
> These aren't meant to be modified. PickedFile represents a file picked from the operating system. If we need to work on a different set of names over the original file then we should use a proper data structure to make this clear.

@lishid But in this case, it would mean just to clone the current PickedFile, changing its props and use it instead of that we got from outside, or not? 

> Also, this wouldn't work if the full path contains a piece that is identical to the enex basename. For example, C:/Evernote/Files/Evernote.enex would only get the first "Evernote" replaced instead of the intended second.

Good point I'll resolve it.

Reply:
> @lishid But in this case, it would mean just to clone the current PickedFile, changing its props and use it instead of that we got from outside, or not?

Can you explain the use case here? If I'm understanding it correctly you'd like to specify a different title/output path for the same input file, right? In that case I would suggest passing this additional information over to the code that uses it (through an object or function parameters) to clearly indicate that it's the supposed output path.

Reply:
@lishid yeah, that would be the use case, and I used to use options object would to store these kind of info. Now I cleaned a code a bit around notebook stacks, hope it looks better for you too.